Makefile

```diff
@@ -8,20 +8,23 @@ REPOSITORY ?= openshift
 TAG ?= latest
 
 LDFLAGS ?= '-extldflags "-static"'
-GO_FILES=$(shell go list ./...)
 
 .DEFAULT_GOAL := help
 
-all: clean verify build test
+all: clean generate verify build test
 .PHONY: all
 
+generate:
+	./hack/update-generated.sh
+.PHONY: generate
+
 test: ## Run unit tests. Example: make test
-	go test $(GO_FILES)
+	go test ./cmd/... ./pkg/...
 .PHONY: test
 
 verify: ## Run verifications. Example: make verify
-	go vet $(GO_FILES)
-	gofmt -w cmd/ pkg/
+	go vet ./cmd/... ./pkg/...
+	gofmt -w ./cmd/ ./pkg/
```

Comment:
I wonder if the "make verify" should also run the "make generate" and verify that the generated files are up to date in the pull request?  I'm not sure if there is a precedent for that in other repos offhand though.

Reply:
Yeah I saw some stuff in openshift/api that might have been applicable if I had ended up pulling that stuff in:
- https://github.com/openshift/api/blob/master/Makefile#L40
- https://github.com/openshift/api/blob/master/Makefile#L43

The deepcopy verify appears to just pass in an additional env var into the existing deepcopy-gen command.  The CRD one pulls down a dev's personal `yq` release and then seems to only do a couple of very basic checks.  I wasn't as enamored with that one at first blush.

Both are things we could pull in but I ended up taking a "meets min" approach with this PR.

Maybe another opportunity for you and me @coreydaley to divde and conquer after this PR merges, where I move on to adding the controller piece to his work (which should not affect the Makefile), while then you wrap up the cobra piece after this merges and then take a stab at verifies?

Reply:
Sounds like a plan to me.